crates/fuel-core/src/database/block.rs

```diff
@@ -82,63 +85,64 @@ impl StorageMutate<FuelBlocks> for Database {
-        });
+    pub fn latest_height(&self) -> DatabaseResult<Option<BlockHeight>> {
+        let id = self.latest_block_ids()?;
         // if no blocks, check if chain was configured with a base height
-        if id.is_none() {
-            id = self.get_starting_chain_height()?;
-        }
+        let id = match id {
+            Some((id, _)) => Some(id),
+            None => self.get_starting_chain_height()?,
+        };
+
         Ok(id)
     }
 
     /// Get the current block at the head of the chain.
     pub fn get_current_block(&self) -> StorageResult<Option<Cow<CompressedBlock>>> {
-        let block_entry = self.latest_block()?;
+        let block_entry = self.latest_block_ids()?;
         match block_entry {
             Some((_, id)) => Ok(StorageAsRef::storage::<FuelBlocks>(self).get(&id)?),
             None => Ok(None),
         }
     }
 
-    pub fn block_time(&self, height: u32) -> StorageResult<Tai64> {
-        let id = self.get_block_id(height.into())?.unwrap_or_default();
+    pub fn block_time(&self, height: BlockHeight) -> StorageResult<Tai64> {
+        let id = self.get_block_id(height)?.unwrap_or_default();
         let block = self
             .storage::<FuelBlocks>()
             .get(&id)?
             .ok_or(not_found!(FuelBlocks))?;
         Ok(block.header().time().to_owned())
     }
 
-    pub fn get_block_id(&self, height: BlockHeight) -> StorageResult<Option<Bytes32>> {
+    pub fn get_block_id(&self, height: BlockHeight) -> StorageResult<Option<BlockId>> {
         Database::get(self, &height.to_bytes()[..], Column::FuelBlockIds)
             .map_err(Into::into)
     }
 
     pub fn all_block_ids(
         &self,
         start: Option<BlockHeight>,
-        direction: Option<IterDirection>,
-    ) -> impl Iterator<Item = DatabaseResult<(BlockHeight, Bytes32)>> + '_ {
+        direction: IterDirection,
+    ) -> impl Iterator<Item = DatabaseResult<(BlockHeight, BlockId)>> + '_ {
         let start = start.map(|b| b.to_bytes().to_vec());
-        self.iter_all::<Vec<u8>, Bytes32>(Column::FuelBlockIds, None, start, direction)
-            .map(|res| {
-                let (height, id) = res?;
-                Ok((
-                    height
-                        .try_into()
-                        .expect("block height always has correct number of bytes"),
-                    id,
-                ))
-            })
+        self.iter_all::<Vec<u8>, BlockId>(
+            Column::FuelBlockIds,
+            None,
+            start,
+            Some(direction),
+        )
+        .map(|res| {
+            let (height, id) = res?;
+            Ok((
+                height
+                    .try_into()
+                    .expect("block height always has correct number of bytes"),
+                id,
+            ))
+        })
     }
 
-    pub fn genesis_block_ids(&self) -> DatabaseResult<(BlockHeight, Bytes32)> {
+    pub fn genesis_block_ids(&self) -> DatabaseResult<(BlockHeight, BlockId)> {
```

Comment:
```suggestion
    pub fn genesis_block_id(&self) -> DatabaseResult<(BlockHeight, BlockId)> {
```

only returns one block id